File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/CollectItemPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record CollectItemPacket(int collectorEntityId, int collectedEntityId) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(collectedEntityId);
        os.writeInt(collectorEntityId);
    }

    @Override
    public byte getId() {
        return 0x0D;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/CollectItemPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class CollectItemPacket(val collectorEntityId: Int, val collectedEntityId: Int) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(collectedEntityId)
        os.writeInt(collectorEntityId)
    }

    override fun getId(): Byte {
        return 0x0D
    }
}
